api/src/messenger.ts

```diff
@@ -38,13 +39,17 @@ export class MessengerRedis implements Messenger {
 	sub: Redis;
 
 	constructor() {
-		const config = getConfigFromEnv('MESSENGER_REDIS');
-
-		this.pub = new IORedis(env.MESSENGER_REDIS ?? config);
-		this.sub = new IORedis(env.MESSENGER_REDIS ?? config);
+		const config = this.getConfig();
+		this.pub = new IORedis(config);
+		this.sub = new IORedis(config);
 		this.namespace = env.MESSENGER_NAMESPACE ?? 'directus';
 	}
 
+	private getConfig() {
+		if ('MESSENGER_REDIS' in env) return env.MESSENGER_REDIS;
+		return merge({}, getConfigFromEnv('REDIS_'), getConfigFromEnv('MESSENGER_REDIS'));
```

Comment:
Off topic from the purpose of this PR, it seems like `MESSENGER_REDIS` (as a connection string) isn't really documented here: https://docs.directus.io/self-hosted/config-options.html#messenger, only `MESSENGER_REDIS_*`.

as apposed to `CACHE_REDIS` and `CACHE_REDIS_*` that are both documented here: https://docs.directus.io/self-hosted/config-options.html#redis-1

Should `MESSENGER_REDIS` get the same "kind" of documentation?

Reply:
> Should `MESSENGER_REDIS` get the same "kind" of documentation?

Yes, it should in my opinion 👍

Reply:
Tracking this in [DOCS-197](https://linear.app/directus/issue/DOCS-197/update-docs-for-messenger-redis-environment-variable) separately 👍